### PR TITLE
Enable adding attributes to generic types

### PR DESCRIPTION
**Pull request by iserrato**
/cc: @jkoritzinsky @agocke

Found while writing Code Fix provider for DynamicallyAccessedMembers trimmer warnings

https://github.com/dotnet/roslyn/issues/63281

### CODE REVIEW

**Comment by jkoritzinsky** [comment]
Since the SyntaxEditor in VB.NET has the same bug, I think we should fix that bug at the same time or soon after. Also, as the bug here is technically in SyntaxGenerator and not SyntaxEditor, I think the following method would be a better location to add the test (though it also works here):

https://github.com/dotnet/roslyn/blob/91aa5211754abfd5441072e8c2ffc0f9a321e300/src/Workspaces/CSharpTest/CodeGeneration/SyntaxGeneratorTests.cs#L1516

The fix and the currently-added test look good to me!

**Review by Youssef1313 — COMMENTED** [review]
Lambdas have the same bug?

File: src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Extensions/SyntaxNodeExtensions.cs [hunk 1]
```diff
@@ -289,6 +289,7 @@ public static SyntaxList<AttributeListSyntax> GetAttributeLists(this SyntaxNode?
                 AccessorDeclarationSyntax accessor => accessor.AttributeLists,
                 ParameterSyntax parameter => parameter.AttributeLists,
                 CompilationUnitSyntax compilationUnit => compilationUnit.AttributeLists,
+                TypeParameterSyntax typeParameter => typeParameter.AttributeLists,
```
Comment (Youssef1313) [inline src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Extensions/SyntaxNodeExtensions.cs:292]: This is missing StatementSyntax as well.

Comment (iserrato) [reply src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Extensions/SyntaxNodeExtensions.cs]: Great catch!

**Comment by iserrato** [comment]
ping @dotnet/roslyn-ide

**Review by CyrusNajmabadi — APPROVED** [review]

**Comment by CyrusNajmabadi** [comment]
thanks!

**Pull request merged**